pinot-query-runtime/src/test/java/org/apache/pinot/query/runtime/QueryRunnerExceptionTest.java

```diff
@@ -71,11 +71,11 @@ private Object[][] provideTestSqlWithExecutionException() {
         // Function with incorrect argument signature should throw runtime exception
         new Object[]{"SELECT least(a.col2, b.col3) FROM a JOIN b ON a.col1 = b.col1",
             "ArithmeticFunctions.least(double,double) with arguments"},
-        // Function that tries to cast String to Number should throw runtime exception
-        new Object[]{"SELECT a.col2, b.col1 FROM a JOIN b ON a.col1 = b.col3", "transform function: cast"},
+        // TODO: this error is thrown but not returned through mailbox. need another test for asserting failure
```

Comment:
this maybe related to 2 issues @61yao filed: https://github.com/apache/pinot/issues/9671 and https://github.com/apache/pinot/issues/9723

after queryplan is generated.
1. QueryDispatcher dispatches the query stages to servers
    - b/c of this behavior, the query dispatch API referred to in #9671 (which is not async) doesn't need to be async since it return immediately after dispatch successful & the GRPC context return immediately after the query plan is valid.
    - however it also should assert initialization of the query plan has no exception, which is what is doesn't do, and what this TODO meant to fix.
2. Server executes the query and returns data via mailboxes
    - once the query plan can be fully translate to an executable operator chain, any error occur during this will be returned by mailbox and it will bubble up eventually. this is what this PR is trying to fix.

Reply:
echo on the 2 issues filed. 
- #9671 dispatch GRPC doesn't need to be async as it doesn't wait for stage complete, it wait for stage validation complete
- #9723 this occurs when mailboxsender or any of the downstream operators throw an exception not in the "nextBlock()" path (for example the init() method). thus the GRPC is opened but not closed.

Reply:
Sorry, just saw this. Can you explain more on why it doesn't need to be async?